protocol/flows/ibd/ibd.go

```diff
@@ -122,17 +122,28 @@ func (flow *handleIBDFlow) downloadBlocks(highestSharedBlockHash *daghash.Hash,
 		return err
 	}
 
+	blocksReceived := 0
 	for {
-		msgIBDBlock, err := flow.receiveIBDBlock()
+		msgIBDBlock, doneIBD, err := flow.receiveIBDBlock()
 		if err != nil {
 			return err
 		}
+
+		if doneIBD {
+			return nil
+		}
+
 		err = flow.processIBDBlock(msgIBDBlock)
 		if err != nil {
 			return err
 		}
-		if msgIBDBlock.BlockHash().IsEqual(peerSelectedTipHash) {
-			return nil
+
+		blocksReceived++
+		if blocksReceived%ibdBatchSize == 0 {
```

Comment:
This can happen even if we are finished with sending.
Instead, I suggest utilize DoneIBDBlocks as you probably planned and then forgot to finish.